### PR TITLE
Fix Deploy action

### PR DESCRIPTION
**Pull request by Discookie**
Uses the Tests action as dependency instead of doing a copy of its own.

### CODE REVIEW

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -6,9 +6,12 @@ on:
     types: [published]
 
 jobs:
+  test:
+    uses: ./.github/workflows/test.yml
   build:
     name: Create extension
     runs-on: ubuntu-18.04
```
Comment (vodorok) [inline .github/workflows/deploy.yml:13]: Github will deprecate the ubuntu-18.04 runner image pretty soon.
 https://github.blog/changelog/2022-08-09-github-actions-the-ubuntu-18-04-actions-runner-image-is-being-deprecated-and-will-be-removed-by-12-1-22/
I thkink you could do the updates in one fell swoop.

Same for the test.yml

**Review by vodorok — CHANGES_REQUESTED** [review]
Please see my remark.

File: .github/workflows/deploy.yml (hunk 1 above)
Comment (Discookie) [reply .github/workflows/deploy.yml]: Updated, and updated the various Actions we're using as well.

Comment (vodorok) [reply .github/workflows/deploy.yml]: Thanks, looks good to me.

**Review by vodorok — APPROVED** [review]
LGTM

**Pull request merged**